docs/source/contributing/development.md

```diff
@@ -0,0 +1,450 @@
+  a custom name for the expected keyword name (if the default is not satisfactory) and
+  additional settings like `tags` and `types`.
+- Hiding keywords logging due to security/privacy reasons can be achieved easily with
+  the `RPA.Robocorp.utils.protect_keywords` utility.
+
+### Library structure
+
+Usually all these methods go into this single main library class, but sometimes you get
+to write a lot of them and is not desirable to manage thousands of lines of code in a
+single module. So you'd normally split the initial module into multiple such modules
+which treat different topics and groups closely related methods into individual keyword
+classes. Then you'd take all these "library parts" and unite them together into the
+main library class.
+
+Some good examples with slightly different approaches:
+- Composition (has-a): [RPA.DocumentAI](../../../packages/main/src/RPA/DocumentAI/DocumentAI.py)
+- Inheritance (is-a): [RPA.Desktop](../../../packages/main/src/RPA/Desktop/__init__.py)
+
+So based on your needs, feel free to choose the simplest and most Pythonic one.
+
+#### Separate package (if needed)
+
+Some libraries are too havy on dependencies and the number of keywords to be
+implemented directly into [main](../../../packages/main), therefore we create a new
+package in this "monolith" repo, and we place them inside by following the same
+structure. Afterwards this becomes self-contained and gets released as a separate
+package into PyPI.
+
+Based on the frequency of usage (and slightly approachable  dependency train), such
+package might get automatically included in the _main_ one  by pinning it into the
+[pyproject.toml](../../../packages/main/pyproject.toml) file. If not, then for sure
+this has to be pinned in the _meta_'s [pyproject.toml](../../../pyproject.toml) in
+order to get documentation built and test how all our packages will go altogether.
+
+### Required changes
+
+Is not easy to know by heart all the changes that need to be done when improving or
+adding a new library, so here's a list with all the areas of interest:
+- The library module/package itself discussed above.
+- Updates into _pyproject.toml_:
+  - Its newly required dependencies should be added with `poetry add <package>` command.
+    (use `-d` option if is a development one)
+    - The new packages will be visible in the project's _pyproject.toml_ file and a
+      command like `inv install.install-updates` will ensure a fresh _poetry.lock_ file
+      after which the current _.venv_ is updated to.
+    - Use `poetry update -vvv` if you want to see the output during the long runs that
+      _meta_ building has.
+  - Bump the `<major>.<minor>.<hotfix>` version number appropriately to reflect the
+    type of change you introduced. (make sure you follow
+    [semantic versioning](https://semver.org/))
+    - The `inv build.version` can help you here with the bumping and unique pre-release
+      version management. (to avoid colision during testing)
```

Comment:
You may want to link the possible bump rules from [here](https://python-poetry.org/docs/cli/#version).